### PR TITLE
Refactor api3 Payment.Get API to support options + most fields in civicrm_financial_trxn

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Payment.get API did not support most of the "options" such as limit/sort etc. Could not search by many of the parameters (eg. order_reference, is_payment etc.).

Before
----------------------------------------
Can't do much with Payment.get API if you don't know the `contribution_id`, `trxn_id` or the `financial trxn ID`.

- `api_v3_PaymentTest::testGetPayment()` tests that `contribution_id` works.
- `api_v3_PaymentTest::testGetPaymentWithTrxnID()` tests that `trxn_id` works.

After
----------------------------------------
Can work with payments using the API (without resorting to FinancialTrxn, EntityFinancialTrxn APIs and lots of wrangling).

Technical Details
----------------------------------------
This API is quite well tested for the things it already does.

Comments
----------------------------------------
@artfulrobot would you be able to review/test this?

### CODE REVIEW

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by jitendrapurohit** [comment]
test this please

**Comment by jitendrapurohit** [comment]
@mattwire Just looked into this and tested by applying the PR and executed Payment APi with few parameters. The option params are working fine but looks like `is_payment` is not returning correct results when set as 0? Eg

    drush cvapi Payment.get sequential=1 contribution_id=99 is_payment=0

and `is_payment=1` returns the same set of results. Can you pls confirm?

**Comment by mattwire** [comment]
@jitendrapurohit I've just pushed a fix for `is_payment` always = 1. Now it works with both.

**Comment by mattwire** [comment]
test this please

File: api/v3/Payment.php [hunk 1]
```diff
@@ -349,6 +364,12 @@ function _civicrm_api3_payment_get_spec(&$params) {
       'type' => CRM_Utils_Type::T_INT,
       'api.aliases' => ['payment_id', 'id'],
     ],
+    'is_payment' => [
```
Comment (eileenmcnaughton) [inline api/v3/Payment.php:367]: Cancel is implemented as a reversal - so I think this is changing the scope & intent of the cancel api in an unintended way

File: api/v3/Payment.php [hunk 2]
```diff
@@ -52,18 +49,36 @@ function civicrm_api3_payment_get($params) {
     if (!empty($eftIds)) {
       $ftParams = [
         'id' => ['IN' => $eftIds],
-        'is_payment' => 1,
+        'is_payment' => $params['is_payment'],
```
Comment (eileenmcnaughton) [inline api/v3/Payment.php:52]: I don't think this change makes sense - the point of the payment api is to retrieve payment financial trxns (payments or refunds or potentially failed payments)

Comment (eileenmcnaughton) [reply api/v3/Payment.php:52]: Note there IS a financial_trxn api for non-payment records

**Comment by eileenmcnaughton** [comment]
@mattwire - I dug into this & came up with an alternate - https://github.com/civicrm/civicrm-core/pull/17071

**Comment by mattwire** [comment]
Merged via #17071

**Pull request closed without merging**